File: code_snippets/python-sdk/workflows/multimodal.py
from clarifai.client.input import Inputs
from clarifai.client.workflow import Workflow

# setting up input image and prompt
prompt = "What time of day is it?"
image_url = "https://samples.clarifai.com/metro-north.jpg"

workflow_url = "https://clarifai.com/clarifai/Visual/workflows/multimodal-to-text"

multi_input = Inputs.get_multimodal_input(input_id="", image_url=image_url, raw_text=prompt)

# Creating the workflow
text_embedding_workflow = Workflow(
    url=workflow_url,
    pat="YOUR_PAT",
)

# Getting the predictions
result = text_embedding_workflow.predict(
    inputs=[
        multi_input
    ]
)
print(result.results[0].outputs[-1].data)

File: code_snippets/python-sdk/workflows/multiModal.ts
import { Input, Workflow } from "clarifai-nodejs";


// Setting up input image and prompt
const prompt = "What time of day is it?";
const imageUrl = "https://samples.clarifai.com/metro-north.jpg";

const workflowUrl =
  "https://clarifai.com/clarifai/Visual/workflows/multimodal-to-text";

const multiInput = Input.getMultimodalInput({
  inputId: "multi-modal-input",
  imageUrl,
  rawText: prompt,
});

// Creating the workflow
const textEmbeddingWorkflow = new Workflow({
  url: workflowUrl,
  authConfig: {
    pat: process.env.CLARIFAI_PAT,
  },
});

// Getting the predictions
const result = await textEmbeddingWorkflow.predict({
  inputs: [multiInput],
});

console.log(
  result.resultsList?.[0]?.outputsList?.[
    result?.resultsList?.[0]?.outputsList?.length - 1
  ]?.data,
);
